backend/api/views/computetask.py

```diff
@@ -91,6 +92,7 @@ def _register_in_orchestrator(tasks_data, channel_name):
         if "__tag__" in orc_task["metadata"]:
             raise Exception('"__tag__" cannot be used as a metadata key')
         else:
+            validate_metadata(orc_task["metadata"])
             orc_task["metadata"]["__tag__"] = task_data.get("tag") or ""
```

Comment:
Here you check for `__` in metadata and add a `__tag__` metadata right after, is it expected ?

Reply:
I've wondered the same thing but from what I saw, this __tag__ as metadata is only in the orchestrator, tag is a field in localrep. So it should be fine since the issue is only in Django.
I'm not sure 100% of this though, I will check further to be sure this tag is never propagated as metadata outside the orchestrator.

Reply:
It makes sense. Something I'm also not comfortable with is re-introducing validation outside of the localrep serializers. If we want this as a rule the two places I would like to see this is in the orchestrator validators for metadata or in the api serializers. (cc @sergebouchut2 @mblottiere)

Reply:
I ruled out doing it in orchestrator because of this `__tag__`, but doing it in the backend it makes totally sense to use the serializer. You're right, thanks for raising the point.
So there is 2 options:
- Option 1 : Use orchestrator validators for metadata, we would need to use another name for the tag than `__tag__`
- Option 2 : Move this check in the serializer (as I should have done right away actually, my bad).

I would favor option 2 as it is a very Django-like issue, but not a strong opinion. I will wait for input from @sergebouchut2 and @mblottiere then before moving forward.

Thanks again!

Reply:
The main issue I can see with option 2 is that the asset will be created in the orchestrator but then rejected by the backend.

Reply:
There's no specific check on the orchestrator about `__tag__`, it's treated as any other metadata and IMO should stay that way.

Reply:
Would it be possible to rename it in the orc and use `_tag_` instead from now on ? We'd need a data migration. We could then do the check that prevents the use of `__` in metadata key, in the orchestrator validators.

Reply:
There is nothing to rename in the orchestrator since the orc doesn't know anything about `__tag__`: this is handled by the backend here: https://github.com/Substra/substra-backend/blob/cb66135ddc9f0c2e3ec45542ffee91d0027c0aa8/backend/orchestrator/client.py#L50-L51

This can be changed, just in mind that this can also impact the SDK/substrafl.

Reply:
As you say in the SDK `tag` is its own field, so I don't see any impact

Reply:
> Thanks for your input, that's good news. We'd still need a data migration on orc side though to avoid dealing with both `__tag__` and `_tag_` in the backend when extracting the tag, right ?

Not sure this should be a data migration (as in, part of the orc): this doesn't concern a field the orc is processing but purely user data. Since filtering is done on API models and not orc data, this should not impact orc data.

Reply:
Check moved to orchestrator in https://github.com/Substra/orchestrator/pull/33
This PR was updated accordingly.